integration_tests/src/main/python/struct_test.py

```diff
@@ -57,3 +57,20 @@ def test_orderby_struct_2(data_gen):
         lambda spark : append_unique_int_col_to_df(spark, unary_op_df(spark, data_gen)),
         'struct_table',
         'select struct_table.a, struct_table.uniq_int from struct_table order by uniq_int')
+
+# conf with legacy cast to string on
+legacy_complex_types_to_string = {'spark.sql.legacy.castComplexTypesToString.enabled': 'true'}
+# @pytest.mark.parametrize('data_gen', [StructGen([["first", boolean_gen], ["second", byte_gen], ["third", short_gen], ["fourth", int_gen], ["fifth", long_gen], ["sixth", string_gen], ["seventh", date_gen], ["eighth", float_gen], ], ["ninth", double_gen], ["tenth", timestamp_gen]])], ids=idfn)
```

Comment:
This set of data gens include `incompat` types (float, double, timestamp), the line below it leaves them out. Because we use a recursive cast to string, we inherit the `incompat` issues and can't test them using a strict cpu and gpu comparison. This seemed like a reasonable way to leave the complete and ideal test set in, so what's actually running isn't accepted as exhaustive.

Reply:
because `data_gen` is parameterized, I personally would prefer a version of it with the failing types marks with xfail and reason on the xfail pointing to the issue(s) that are causing it to fail. I am fine with the commented out code so long as it also has a comment explaining why it is commented out, and ideally a link to the issue(s) that need to be fixed for it.

Reply:
Changed to xfail for the 3 failing types.